src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -43,26 +41,28 @@ public EditCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
         }
 
-        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
-        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
-            editPersonDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
-        }
-        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
-            editPersonDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
-        }
-        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
-            editPersonDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
-        }
-        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
-            editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
-        }
-        parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editPersonDescriptor::setTags);
+        String itemType = ParserUtil.parseItemType(argMultimap.getValue(PREFIX_ITEM).get());
+
+        switch (itemType) {
+        case "res":
+            EditResumeDescriptor editResumeDescriptor = new EditResumeDescriptor();
+            // TODO: Not sure if I should create a method inside the respective descriptors for this checking.
+            if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
```

Comment:
Not sure if I understand argMultimap correctly, but maybe you can do the checking outside of the switch statements? so the logic will be:
check if name is present
--> switch cases to create the correct editDescriptor
--> make the necessary changes (setName, setTags...)
--> throw exception if no field edited